portfolio/src/main/webapp/script.js

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Fetches content of /data from the server and adds it to the DOM.
+ */
+function getData() {
+  console.log('Fetching data.');
+
+  const responsePromise = fetch('/data');
+
+  responsePromise.then(handleResponse);
+}
+
+/**
+ * Handles response by converting it to text and passing the result to
+ * addDataToDom().
+ */
+function handleResponse(response) {
+  console.log('Handling the response.');
+
+  const textPromise = response.text();
+
+  textPromise.then(addDataToDom);
```

Comment:
another way to write these, that personally makes a LOT more sense to me when trying to understand the flow is to use javascripts async/await .

https://javascript.info/async-await

Reply:
Hi Isis, you can also check out the following example from the STEP project:
https://github.com/googleinterns/step/blob/640ff5d6c6484da82f1494d04d0490861c696fa6/walkthroughs/week-3-server/examples/random-quotes/src/main/webapp/script.js#L69

If you prefer, I think it would be okay to keep things are they are, and update the code later to use async/await.